sources/assets/Stride.Core.Assets/PackageSession.Dependencies.cs

```diff
@@ -132,6 +136,18 @@ private async Task PreLoadPackageDependencies(ILogger log, SolutionProject proje
                             if (referencedProject != null)
                             {
                                 await PreLoadPackageDependencies(log, referencedProject, loadParameters);
+
+                                // Get package upgrader from dependency (a project might depend on another project rather than referencing Stride directly)
+                                // A better system would be to evaluate nuget flattened dependencies WITHOUT doing the actual restore (dry-run).
+                                // However I am not sure it's easy/possible to do it (using API) without doing a full restore/download, which we don't want to do
+                                // with old version (it might be uninstalled already and we want to avoid re-downloading it again)
+                                pendingPackageUpgradesPerPackage.TryGetValue(referencedProject.Package, out var dependencyPackageUpgraders);
+                                foreach (var dependencyPackageUpgrader in dependencyPackageUpgraders)
```

Comment:
Sorry if I'm jumping the gun and this is still a work in progress.
I've created a new SpriteStudio sample project and it crashed because `dependencyPackageUpgraders` is null, because the TryGetValue directly above it failed. Wrapping it in an if statement allowed it to go through and the project is created without any issues.

Reply:
Good point, something I overlooked when doing recent changes (I guess I was only testing upgrades, and didn't try new projects again yet).
Fixed with latest commit, thanks!